src/destructuring.rs

```diff
@@ -12,14 +12,24 @@ use crate::{
     types::{TypeF, Types},
 };
 
+#[derive(Debug, PartialEq, Clone)]
+pub enum FieldPattern {
+    /// An assignment match like `{ ..., a = b, ... }`
+    Ident(Ident),
+    /// A nested record pattern like `{ ..., a = { b, c }, ... }`
+    RecordPattern(RecordPattern),
+    /// An aliased nested record pattern like `{ ..., a = b @ { c, d }, ... }`
+    AliasedRecordPattern(Ident, RecordPattern),
```

Comment:
Nitpick: any reason not to use a struct here? `AliasedRecordPattern { alias: Ident, pattern: RecordPattern }`